Login update

wulalasuperdope opened the pull request:
See [comment](https://github.com/1909-capstone/climbAR/pull/50#issuecomment-596869293) below for additional changes.

- fetch this pr
- run the app
- go to Home, Climbing Routes page or Sign up page
- log in, you should be able to get back the previously visited page

Issues:
1. I'm having problem to setting condition to redirect after logged in: if the previously visited page is Sign up, user should not go back to Sign up page.

TO DO:
- [ ] (not necessary) create a component for "Not Found, Redirecting to /login' page
- [x]  redirect Admin to Home page when he logged out

wulalasuperdope committed:
save the work

wulalasuperdope committed:
fixed user auth for log in

wulalasuperdope committed:
deleted comments

nries1 commented:
Taking a look now.

Checks:
- [x]   When I log in, I get redirected to the last page I visited.

Changes:
- [x] I redirected the user to their last visited page after signing up. 
- [x] I redirected the user to 404 page when user visits an admin only page.
- [x] Problem: When you log in, we send the user back to their last visited page with vanilla `window.back()`. That doesn't work anymore b/c the previous page might be a 404 and we don't want to send users there.
Solution: Don't use vanilla `window.back()` to navigate the user after log in. Instead use the React history API. Note, for now we are just redirecting to the home page. In the future we might be able to redirect them to the last page visited but we can't force a page refresh we instead need to use react router history.
- [x] When users are on Create Routes, then log out, they are redirected to the home page.

wulalasuperdope committed:
fixed redirecting when user has a cookie but not logged in yet

nries1 committed:
redirect after signup

wulalasuperdope committed:
merging origin branch
Merge branch 'login-update' of https://github.com/1909-capstone/climbAR into login-update

nries1 committed:
redirect to 404 page when user visites a whitelisted page without a valid cookie

nries1 commented on server/server.js:
```diff
@@ -50,17 +50,16 @@ app.use((req, res, next) => {
+          console.log(chalk.cyan('user is not found'));
           //status: user has a cookie id, but login expired
           //next step: user can view pages except Create Route or profile
-          if (whiteList[req.path] === false && user.userType !== 'Admin') {
```
had to remove the user.userType condition because if the middleware reaches this point, then user is null

wulalasuperdope replied:
Yeah I got this as well, just a few seconds slower than you ;D

nries1 replied:
xD

I'm working on making it so that the page doesn't redirect to login when an admin goes to admin/create -> login -> admin/create

wulalasuperdope committed:
merged with origin login-update branch

wulalasuperdope committed:
converted seeded position to integer (Oops, should have started a new branch)

nries1 committed:
used react history and Redirect component to redirect users home when they login or signup

nries1 committed:
used react history and Redirect component to redirect users home when they login or signup

wulalasuperdope commented:
> Taking a look now.
> 
> Checks:
> 
> * [x]    When I log in, I get redirected to the last page I visited.
> 
> Changes:
> 
> * [x]  I redirected the user to their last visited page after signing up.
> * [x]  I redirected the user to 404 page when user visits an admin only page.
> * [x]  Problem: When you log in, we send the user back to their last visited page with vanilla `window.back()`. That doesn't work anymore b/c the previous page might be a 404 and we don't want to send users there.
>   Solution: Don't use vanilla `window.back()` to navigate the user after log in. Instead use the React history API. Note, for now we are just redirecting to the home page. In the future we might be able to redirect them to the last page visited but we can't force a page refresh we instead need to use react router history.

Sure redirecting to the home page is good to me. Thanks for helping!

nries1 committed:
redirect users to login page when they are on create route and they log out

wulalasuperdope commented:
> Taking a look now.
> 
> Checks:
> 
> * [x]    When I log in, I get redirected to the last page I visited.
> 
> Changes:
> 
> * [x]  I redirected the user to their last visited page after signing up.
> * [x]  I redirected the user to 404 page when user visits an admin only page.
> * [x]  Problem: When you log in, we send the user back to their last visited page with vanilla `window.back()`. That doesn't work anymore b/c the previous page might be a 404 and we don't want to send users there.
>   Solution: Don't use vanilla `window.back()` to navigate the user after log in. Instead use the React history API. Note, for now we are just redirecting to the home page. In the future we might be able to redirect them to the last page visited but we can't force a page refresh we instead need to use react router history.
> * [x]  When users are on Create Routes, then log out, they are redirected to the home page.

I was redirected to login page after logging out from Create Routes.

nries1 committed:
restored link styling after removed all of the href attributes

nries1 commented:
Thanks @wulalasuperdope . I think this is ready for review.

wulalasuperdope commented:
> Thanks @wulalasuperdope . I think this is ready for review.

nries1 commented:
@wulalasuperdope  are you not able to merge?

wulalasuperdope commented on app/components/Navigation.js:
```diff
@@ -30,9 +39,7 @@ class Navigation extends Component {
-              <NavDropdown.Item eventKey="4.1" href={`/profile`}>
-                Profile
-              </NavDropdown.Item>
+              <NavDropdown.Item to={`/profile`}>Profile</NavDropdown.Item>
```
the link to Profile page is broken. I'll take a look at the link to NavDropDown.Item tmr.

nries1 replied:
Oh I see. It looks like NavDropdown.Item doesn't support a to attribute. It wants href. But since we can't use href, we're going to need to use a different html element here

nries1 replied:
Just fixed it.

wulalasuperdope reviewed (commented):
Nice job! just the profile link needs to be fixed.

nries1 committed:
restored to attribute to profile link

nries1 committed:
Merge branch 'dev' into login-update
